clients/interfaces/event.go

```diff
@@ -31,13 +33,17 @@ type EventClient interface {
 	// offset: The number of items to skip before starting to collect the result set. Default is 0.
 	// limit: The number of items to return. Specify -1 will return all remaining items after offset. The maximum will be the MaxResultCount as defined in the configuration of service. Default is 20.
 	EventsByDeviceName(ctx context.Context, name string, offset, limit int) (responses.MultiEventsResponse, errors.EdgeX)
+	// EventsByDeviceNameWithQueryParams returns a portion of the entire events according to the device name and specified query parameters. Events are sorted in descending order of created time.
+	EventsByDeviceNameWithQueryParams(ctx context.Context, name string, queryParams map[string]string) (responses.MultiEventsResponse, errors.EdgeX)
```

Comment:
```suggestion
	EventsByDeviceNameWithQueryParams(ctx context.Context, name string, offset, limit int, queryParams map[string]string) (responses.MultiEventsResponse, errors.EdgeX)
```

Reply:
The offset and limit are part of the query parameter, so I remove offset, limit int from the function argument.
For example, http://0.0.0.0:59880/api/v3/event/device/name/Random-UnsignedInteger-Device?offset=0&limit=3&numeric=true

Reply:
Added the offset, limit back.